src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsSquiggleTaggerProvider.TagSource.cs

```diff
@@ -39,7 +39,8 @@ protected override int MinimumLength
             protected override bool ShouldInclude(DiagnosticData diagnostic)
             {
                 return
-                    (diagnostic.Severity == DiagnosticSeverity.Warning || diagnostic.Severity == DiagnosticSeverity.Error) &&
+                    (diagnostic.Severity == DiagnosticSeverity.Warning || diagnostic.Severity == DiagnosticSeverity.Error ||
+                    (diagnostic.Severity == DiagnosticSeverity.Hidden && diagnostic.CustomTags.Contains((c) => c == WellKnownDiagnosticTags.Unnecessary))) &&
```

Comment:
Can we add an overload of `Contains` (or is there already?) that takes the value searched for rather than a predicate. I guess it could save some allocations.